lnwallet/channel.go

```diff
@@ -5107,10 +5143,18 @@ func (lc *LightningChannel) validateFeeRate(feePerKw btcutil.Amount) error {
 	newFee := lnwire.NewMSatFromSatoshis(
 		btcutil.Amount((int64(feePerKw) * txWeight) / 1000),
 	)
-	balanceAfterFee := availableBalance - newFee
+
+	// If the total fee exceeds our available balance, then we'll reject
+	// this update as it would mean we need to trim our entire output.
+	if newFee > availableBalance {
```

Comment:
👍

Reply:
This can be skipped by making the below check `if availableBalance.toSatoshis() < lc.channelState.LocalChanCfg.ChanReserve + newFee` and keeping `newFee` in satoshis instead of msat.

Reply:
Sure it can be skipped, but IMO your suggested replacement isn't as intuitive as just checking the fee immediately after computing it.